src/js/AppStore.js

```diff
@@ -0,0 +1,196 @@
+import React from 'react';
+import Modal from 'react-modal'
+import { connect } from 'react-redux';
+import store from './store';
+import { flags } from './Flags'
+
+import AppHeader from './containers/Header';
+import HScrollMenu from './HScrollMenu';
+import AppServices from './containers/AppServices';
+import ConfirmAlert from './ConfirmAlert';
+
+import BCController from './Controllers/BCController';
+import fileSystemController from './Controllers/FileSystemController'
+
+import { updateAvailableAppStoreApps, appStoreAddAppPendingSetAppProperties } from './actions';
+
+class AppStore extends React.Component {
+    constructor() {
+        super();
+
+        this.state = {
+            confirmID: null,
+            confirmApp: null
+        }
+
+        this.cancel = this.cancel.bind(this);
+        this.confirm = this.confirm.bind(this);
+        this.onSelection = this.onSelection.bind(this);
+    }
+
+    cancel() {
+        this.setState((state, props) => {
+            return {
+                confirmID: null,
+                confirmApp: null
+            }
+        });
+    }
+
+    confirm() {
+        fileSystemController.subscribeToEvent('InstallApp', (success, params) => {
+            if (!success || !params.appUrl) {
+                console.error('error encountered when installing app');
+            }
+
+            fetch(params.appUrl + 'manifest.js')
+                .then(x => x.text())
+                .then((manifestJS) => {
+                let jsonStart = manifestJS.indexOf('{');
+                let jsonEnd = manifestJS.lastIndexOf('}') + 1;
+                var manifest = {};
+
+                try {
+                    manifest = JSON.parse(manifestJS.substring(jsonStart, jsonEnd));
+                } catch (e) {
+                    console.error('failed to parse manifest as JSON: ', e);
+                    return;
+                }
+
+                var appProperties = {
+                    policyAppID: manifest.appId,
+                    enabled: true
+                }
+
+                let state = store.getState();
+                var appDirEntry = state.appStore.availableApps.find(x => x.policyAppID == manifest.appId);
+
+                if (!appDirEntry) {
+                    console.error('no app found in app store directory with policyAppID ', manifest.appId);
+                    return;
+                }
+
+                store.dispatch(appStoreAddAppPendingSetAppProperties(Object.assign(appDirEntry, { 
```

Comment:
I would remove the appStore prefixes from all of the app store related actions. The name is so long

Reply:
I renamed this action in https://github.com/smartdevicelink/generic_hmi/pull/240/commits/0f726e440962d70e67026eaa39e726cd0e342bf4, but I found the other action names to be reasonable. Let me know if you think appStore in any of the other names is redundant.